Source/Plugins/EditorModules/ObjectInspector/Modules/ObjectInspector.cs

```diff
@@ -53,9 +54,36 @@ public bool EmptySelection
 		public ObjectInspector(int runtimeId)
 		{
 			this.InitializeComponent();
+			this.buttonLock.CheckedChanged += (sender, args) => this.UserSettings.Locked = this.buttonLock.Checked;
 			this.runtimeId = runtimeId;
 			this.toolStrip.Renderer = new Duality.Editor.Controls.ToolStrip.DualitorToolStripProfessionalRenderer();
 		}
+		
+		public void ApplyUserSettings()
+		{
+			this.buttonAutoRefresh.Checked = this.userSettings.AutoRefresh;
+			this.buttonLock.Checked = this.userSettings.Locked;
+			this.buttonDebug.Checked = this.userSettings.DebugMode;
+			this.buttonSortByName.Checked = this.userSettings.SortByName;
+			this.Text = this.userSettings.TitleText;
+
+			if (this.userSettings.ExpandState != null)
+			{
+				this.gridExpandState.LoadFromXml(this.userSettings.ExpandState);
+			}
+		}
+
+		internal void SaveGridExpandState()
```

Comment:
Not sure if this is right.

Reply:
Can you go into detail why? I think I'm missing context.